focuswriter: 1.7.6 -> 1.8.5

kashw2 opened the pull request:
## Description of changes

change log: https://github.com/gottcode/focuswriter/releases/tag/v1.8.5

diff: https://github.com/gottcode/focuswriter/compare/v1.7.6...v1.8.5

- added myself as a maintainer of focuswriter
- changed `all-packages.nix` call to focuswriter to use `qt6Packages.callPackage` over `libsForQt5.callPackage` as focuswriter now uses qt6 predominantly. 
- swapped out `fetchurl` to `fetchFromGitHub` as the GitHub acts as a sort of mirror and has interop with `nix-update` for ease of updates in the future
- wrapped application

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

kashw2 commented:
Result of `nixpkgs-review pr 253111` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>focuswriter</li>
  </ul>
</details>

NickCao reviewed (changes requested):
Hash is not updated, also please split the maintainer addition into its own commit.

kashw2 committed:
focuswriter: 1.7.6 -> 1.8.5

kashw2 committed:
focuswriter: added kashw2 maintainer

kashw2 committed:
focuswriter: switch from `fetchurl` to `fetchFromGitHub`

kashw2 commented:
Hi @NickCao. Thank you for catching this.

I've actually been given a good opportunity to learn more about qt6 and packaging qt apps for Nix due to this which I appreciate. 

I've added a few more commits. I've updated the description of the Pull Request to document the changes and their rationales.

Thanks.

kashw2 commented:
Result of `nixpkgs-review pr 253111` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>focuswriter</li>
  </ul>
</details>

NickCao commented on pkgs/applications/editors/focuswriter/default.nix:
```diff
@@ -1,24 +1,37 @@
+  nativeBuildInputs = [ pkg-config cmake qttools ];
+  buildInputs = [ hunspell qtbase qtmultimedia qt5compat ];
+
+  dontWrapQtApps = true;
```
Why aren't we wrapping it?

kashw2 replied:
I'd had issues early on when attempting to wrap it however it looks like those are gone now.

fixed in https://github.com/NixOS/nixpkgs/pull/253111/commits/1f95a84bdaff12b56bcb653c9251dba6206c8f70

NickCao commented on pkgs/applications/editors/focuswriter/default.nix:
```diff
@@ -1,24 +1,37 @@
-  nativeBuildInputs = [ pkg-config qmake qttools ];
-  buildInputs = [ hunspell qtbase qtmultimedia ];
+  nativeBuildInputs = [ pkg-config cmake qttools ];
+  buildInputs = [ hunspell qtbase qtmultimedia qt5compat ];
```
Also please add `qtwayland` for, wayland support.

kashw2 replied:
added qtwayland to `buildInputs` in https://github.com/NixOS/nixpkgs/pull/253111/commits/1f95a84bdaff12b56bcb653c9251dba6206c8f70

kashw2 committed:
focuswriter: added qtwayland and wrapped application

NickCao reviewed (approved):
diff lgtm

NickCao commented:
Tested building, running and typing a few words.
